modules/context/access_log.go

```diff
@@ -20,20 +22,51 @@ type routerLoggerOptions struct {
 	Start          *time.Time
 	ResponseWriter http.ResponseWriter
 	Ctx            map[string]interface{}
+	RequestID      *string
 }
 
 var signedUserNameStringPointerKey interface{} = "signedUserNameStringPointerKey"
 
+const keyOfRequestIDInTemplate = ".RequestID"
+
+// According to:
+// In OpenTracing and OpenTelemetry, the maximum length of trace id is 256 bits (32 bytes).
+// MD5 output is 16 or 32 bytes.
+// UUID output is 36 bytes (including four ‘-’)
+// SHA1 output is 40 bytes
+// So, we accept a Request ID with a maximum character length of 40
+const maxRequestIDBtyeLength = 40
+
+func parseRequestIDFromRequestHeader(req *http.Request) string {
+	requestID := "-"
+	for _, key := range setting.RequestIDHeaders {
+		if req.Header.Get(key) != "" {
+			requestID = req.Header.Get(key)
+			break
+		}
+	}
+	if len(requestID) > maxRequestIDBtyeLength {
+		requestID = fmt.Sprintf("%s...", requestID[:maxRequestIDBtyeLength])
+	}
+	return requestID
+}
+
 // AccessLogger returns a middleware to log access logger
 func AccessLogger() func(http.Handler) http.Handler {
 	logger := log.GetLogger("access")
 	logTemplate, _ := template.New("log").Parse(setting.AccessLogTemplate)
+	needRequestID := strings.Contains(setting.AccessLogTemplate, keyOfRequestIDInTemplate)
```

Comment:
Could be cheated by a template like `abc.RequestIDefg {{.Ctx.RemoteAddr}} - {{.Identity}} {{.Start.Format "[02/Jan/2006:15:04:05 -0700]" }} ...`, but I don't think it's worth doing something like https://stackoverflow.com/questions/40584612/how-to-get-a-map-or-list-of-template-actions-from-a-parsed-template

 The impact is just a little waste of time trying to get the request id which is not needed, so I think it's OK.

Reply:
Maybe:
```suggestion
	needRequestID := len(setting.RequestIDHeaders) > 0 && strings.Contains(setting.AccessLogTemplate, keyOfRequestIDInTemplate)
```